### PR TITLE
[WIP] chore(update): postcss-selector-parser

### PR DESCRIPTION
**Pull request by evilebottnawi**
> Which issue, if any, is this issue related to?

update parser

> Is there anything in the PR that needs further explanation?

Maybe refactor code base, in some places we can improve speed add more check or simplify code, but it is out of this PR

Very few problems

### CODE REVIEW

File: lib/utils/isStandardSyntaxCombinator.js [hunk 1]
```diff
@@ -8,24 +8,13 @@
...
-  ) {
-    return false;
-  }
-
-  return true;
+  return (
+    node.type === "combinator" &&
+    !node.value.startsWith("/") &&
+    !node.value.endsWith("/")
+  );
```
Comment (evilebottnawi) [inline lib/utils/isStandardSyntaxCombinator.js]: No need lodash, we already require node@6, so we can use this function

File: lib/rules/string-quotes/index.js [hunk 2]
```diff
@@ -75,52 +75,97 @@ const rule = function(expectation, secondary, context) {
...
-                openIndex +
-                // the length of our value
-                attributeNode.value.length -
-                // with the length of our quote subtracted
-                erroneousQuote.length;
-
-              fixPositions.push(openIndex, closeIndex);
+            if (context.fix) {
+              selectorFixed = true;
+              attributeNode.quoteMark = correctQuote;
```
Comment (evilebottnawi) [inline lib/rules/string-quotes/index.js]: Port from https://github.com/stylelint/stylelint/pull/3284

File: lib/rules/selector-pseudo-class-parentheses-space-inside/__tests__/index.js [hunk 3]
```diff
@@ -312,13 +313,13 @@ testRule(rule, {
...
-      fixed: "section:not(/**/ :has(/**/ h1, h2 /**/) /**/) { }",
+    }
+    /*{
+      code: "section:not( /!**!/ :has( /!**!/ h1, h2 /!**!/ ) /!**!/ ) { }",
+      fixed: "section:not(/!**!/ :has(/!**!/ h1, h2 /!**!/) /!**!/) { }",
       message: messages.rejectedOpening,
       line: 1,
       column: 13
-    }
+    }*/
```
Comment (evilebottnawi) [inline lib/rules/selector-pseudo-class-parentheses-space-inside/__tests__/index.js]: Looks like a bug in parser, investigate

File: lib/rules/selector-descendant-combinator-no-non-space/__tests__/index.js [hunk 4]
```diff
@@ -111,12 +111,12 @@ testRule(rule, {
...
     {
-      code: ".foo /*comment*/  /*comment*/ .bar > /*comment*/   .buz {}",
-      fixed: ".foo /*comment*/ /*comment*/ .bar > /*comment*/   .buz {}",
+      code: ".foo /!*comment*!/  /!*comment*!/ .bar > /!*comment*!/   .buz {}",
+      fixed: ".foo /!*comment*!/ /!*comment*!/ .bar > /!*comment*!/   .buz {}",
       message: messages.rejected("  "),
       line: 1,
       column: 17
-    },
+    },*/
```
Comment (evilebottnawi) [inline lib/rules/selector-descendant-combinator-no-non-space/__tests__/index.js]: Looks like a bug in parser, investigate

File: system-tests/005/__snapshots__/005.test.js.snap [hunk 5]
```diff
@@ -16149,22 +16149,6 @@ Array [
...
-      },
-      Object {
-        "column": 2,
-        "line": 188,
-        "rule": "selector-descendant-combinator-no-non-space",
-        "severity": "error",
-        "text": "Unexpected \\"
-
-\\" (selector-descendant-combinator-no-non-space)",
-      },
```
Comment (evilebottnawi) [inline system-tests/005/__snapshots__/005.test.js.snap]: Looks like a bug in parser, investigate

**Comment by evilebottnawi** [comment]
Still buggy and very misleading in raw/spaces, we need fix it in `postcss-value-parser`, WIP

**Comment by evilebottnawi** [comment]
Blocker https://github.com/postcss/postcss-selector-parser/issues/189

**Comment by jeddy3** [comment]
@evilebottnawi Thanks for working on this!

Is this still blocked by postcss/postcss-selector-parser#189?

**Comment by evilebottnawi** [comment]
Yep blocked, we need again do major release for selector parser, I think it was done to end of month

**Comment by jeddy3** [comment]
> Yep blocked, we need again do major release for selector parser

Do you mean a major release of the selector parser or a major release of stylelint?

**Comment by evilebottnawi** [comment]
Major release parser

**Comment by jeddy3** [comment]
Can we close #3861 in favour of this PR?

**Comment by evilebottnawi** [comment]
@jeddy3 yes

**Comment by hudochenkov** [comment]
@evilebottnawi what prevents us update to the latest version?

**Comment by evilebottnawi** [comment]
@hudochenkov need fix comment parsing, i don't have enough time to fix it, sorry, maybe in future

**Comment by hudochenkov** [comment]
@evilebottnawi is this fix should be made in parser or in stylelint?

**Comment by evilebottnawi** [comment]
@hudochenkov parser

**Comment by jeddy3** [comment]
Closed in favour of https://github.com/stylelint/stylelint/pull/4595

**Pull request closed without merging**